src/utils/searchViewHelper.js

```diff
@@ -20,10 +26,19 @@ const searchViewHelper = {
       store.dispatch(fetchRevisionsByAuthor({ repository, search }));
     } else if (longHashMatch.test(search) || shortHashMatch.test(search)) {
       store.dispatch(fetchRevisionByID({ repository, search }));
+    } else {
+      store.dispatch(updateInputError(true));
+      store.dispatch(
+        updateInputHelperText(
+          'Search must be a 12- or 40-character hash, or email address',
+        ),
+      );
```

Comment:
I feel like the same action could be used to change these 2 values, because they're linked.

```js
store.dispatch(setInputInError(errorReason))
```
would set `inputError` to `true` and set the `errorReason` together.
and

```js
store.dispatch(setInputNoError())
```
would set `inputError` to `false` and set the `errorReason` to `''`.

In general it's a good idea to reason about redux actions as user actions, rather as a way to update each individual state properties. That's the reducer's job to update the state from a logical redux action.

Don't worry, this is a quite common error, and it's too late to correct it :-)